satpy/readers/li_l2_nc.py

```diff
@@ -158,11 +158,9 @@ def get_array_on_fci_grid(self, data_array: xr.DataArray):
         data_2d = da.map_blocks(_np_add_at_wrapper, data_2d, (rows, cols), data_array,
                                 dtype=data_array.dtype,
                                 chunks=(LI_GRID_SHAPE[0], LI_GRID_SHAPE[1]))
-        data_2d = da.where(data_2d > 0, data_2d, np.nan)
-
+        data_2d = da.where(data_2d > 0, data_2d, np.nan).astype(np.float32)

```

Comment:
This would prevent one unnecessary upcasting:
```suggestion
        data_2d = da.where(data_2d > 0, data_2d, np.float32(np.nan))

```

Reply:
@pnuu  casting a `nan` who is a float32 is not enough to convert all the arrays into a `float32` . For example if `data_2d` is an `int32` the method where will convert it to a `float64`. To prevent it I have used the method `astype(np.float32)`.